router/dataplane.go

```diff
@@ -2157,23 +2195,27 @@ func (p *scionPacketProcessor) validateEgressID() (respr processResult, reserr e
 			cannotRoute,
 		)
 	}
-
 	// @ p.d.getLinkTypesMem()
 	ingress, egress := p.d.linkTypes[p.ingressID], p.d.linkTypes[pktEgressID]
 	if !p.segmentChange {
 		// Check that the interface pair is valid within a single segment.
 		// No check required if the packet is received from an internal interface.
+		// @ assert !p.segmentChange
```

Comment:
I would drop sth like this, which is established by the path condition just before